File: frontend/android/app/src/main/kotlin/com/example/imagetoolkit/ui/screen/DatabaseScreen.kt
// ui/screen/DatabaseScreen.kt
package com.example.imagetoolkit.ui.screen

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Power
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.unit.dp
import com.example.imagetoolkit.ui.components.SectionCard
import android.widget.Toast

@OptIn(ExperimentalMaterial3Api::class, ExperimentalLayoutApi::class)
@Composable
fun DatabaseScreen() {
    var host by remember { mutableStateOf("localhost") }
    var port by remember { mutableStateOf("5432") }
    var user by remember { mutableStateOf("postgres") }
    var password by remember { mutableStateOf("") }
    var dbName by remember { mutableStateOf("image_db") }
    var imagePath by remember { mutableStateOf("") }
    var seriesName by remember { mutableStateOf("") }
    var seriesExpanded by remember { mutableStateOf(false) }
    var characters by remember { mutableStateOf("") }
    var selectedTags by remember { mutableStateOf(setOf<String>()) }
    val allTags = listOf(
        "landscape", "night", "day", "indoor", "outdoor", "solo", "multiple", "fanart",
        "official", "cosplay", "portrait", "full_body", "action", "close_up", "nsfw"
    )
    val seriesOptions = listOf("Naruto", "Bleach", "One Piece", "Dragon Ball")
    val context = LocalContext.current

    LazyColumn(
        modifier = Modifier.fillMaxSize().padding(16.dp),
        verticalArrangement = Arrangement.spacedBy(16.dp)
    ) {
        item {
            SectionCard(title = "PostgreSQL Connection", startOpen = true) {
                Column(verticalArrangement = Arrangement.spacedBy(8.dp)) {
                    OutlinedTextField(value = host, onValueChange = { host = it }, label = { Text("Host") }, modifier = Modifier.fillMaxWidth())
                    OutlinedTextField(value = port, onValueChange = { port = it }, label = { Text("Port") }, keyboardOptions = androidx.compose.ui.text.input.KeyboardOptions(keyboardType = androidx.compose.ui.text.input.KeyboardType.Number), modifier = Modifier.fillMaxWidth())
                    OutlinedTextField(value = user, onValueChange = { user = it }, label = { Text("User") }, modifier = Modifier.fillMaxWidth())
                    OutlinedTextField(value = password, onValueChange = { password = it }, label = { Text("Password") }, visualTransformation = PasswordVisualTransformation(), modifier = Modifier.fillMaxWidth())
                    OutlinedTextField(value = dbName, onValueChange = { dbName = it }, label = { Text("Database Name") }, modifier = Modifier.fillMaxWidth())
                    Button(
                        onClick = { Toast.makeText(context, "Connecting to $dbName...", Toast.LENGTH_SHORT).show() },
                        modifier = Modifier.fillMaxWidth()
                    ) {
                        Icon(Icons.Default.Power, contentDescription = null, modifier = Modifier.padding(end = 8.dp))
                        Text("Connect to PostgreSQL")
                    }
                }
            }
        }

        item {
            Text(
                "Stats: Not connected to database",
                modifier = Modifier
                    .fillMaxWidth()
                    .background(MaterialTheme.colorScheme.surfaceVariant, RoundedCornerShape(8.dp))
                    .padding(16.dp),
                color = MaterialTheme.colorScheme.onSurfaceVariant
            )
        }

        item {
            SectionCard(title = "Single Image Metadata", startOpen = true) {
                Column(verticalArrangement = Arrangement.spacedBy(16.dp)) {
                    OutlinedTextField(
                        value = imagePath,
                        onValueChange = { imagePath = it },
                        label = { Text("Image File Path") },
                        modifier = Modifier.fillMaxWidth()
                    )
                    Row(horizontalArrangement = Arrangement.spacedBy(8.dp)) {
                        Button(onClick = { /* Browse */ }, modifier = Modifier.weight(1f)) { Text("Browse") }
                        Button(onClick = { /* View */ }, modifier = Modifier.weight(1f)) { Text("View") }
                        Button(onClick = { /* Load */ }, modifier = Modifier.weight(1f)) { Text("Load") }
                    }

                    Box(
                        modifier = Modifier
                            .fillMaxWidth()
                            .height(200.dp)
                            .background(MaterialTheme.colorScheme.secondaryContainer, RoundedCornerShape(8.dp)),
                        contentAlignment = Alignment.Center
                    ) {
                        Text("Select an image to edit...", color = MaterialTheme.colorScheme.onSecondaryContainer)
                    }

                    ExposedDropdownMenuBox(
                        expanded = seriesExpanded,
                        onExpandedChange = { seriesExpanded = !seriesExpanded }
                    ) {
                        OutlinedTextField(
                            value = seriesName,
                            onValueChange = { seriesName = it },
                            label = { Text("Series Name") },
                            trailingIcon = { ExposedDropdownMenuDefaults.TrailingIcon(expanded = seriesExpanded) },
                            modifier = Modifier.fillMaxWidth().menuAnchor()
                        )
                        ExposedDropdownMenu(
                            expanded = seriesExpanded,
                            onDismissRequest = { seriesExpanded = false }
                        ) {
                            seriesOptions.forEach { series ->
                                DropdownMenuItem(
                                    text = { Text(series) },
                                    onClick = {
                                        seriesName = series
                                        seriesExpanded = false
                                    }
                                )
                            }
                        }
                    }

                    OutlinedTextField(
                        value = characters,
                        onValueChange = { characters = it },
                        label = { Text("Characters (comma-separated)") },
                        modifier = Modifier.fillMaxWidth()
                    )

                    Text("Tags", style = MaterialTheme.typography.titleSmall)
                    FlowRow(
                        modifier = Modifier.fillMaxWidth(),
                        horizontalArrangement = Arrangement.spacedBy(8.dp),
                        verticalArrangement = Arrangement.spacedBy(4.dp)
                    ) {
                        allTags.forEach { tag ->
                            val isSelected = selectedTags.contains(tag)
                            FilterChip(
                                selected = isSelected,
                                onClick = {
                                    selectedTags = if (isSelected) selectedTags - tag else selectedTags + tag
                                },
                                label = { Text(tag) }
                            )
                        }
                    }

                    Column(verticalArrangement = Arrangement.spacedBy(8.dp)) {
                        Button(onClick = { /* Add/Update */ }, modifier = Modifier.fillMaxWidth()) {
                            Text("Add/Update Image Path")
                        }
                        Button(onClick = { /* Update Meta */ }, modifier = Modifier.fillMaxWidth()) {
                            Text("Update Loaded Metadata")
                        }
                        Button(
                            onClick = { /* Delete */ },
                            modifier = Modifier.fillMaxWidth(),
                            colors = ButtonDefaults.buttonColors(containerColor = MaterialTheme.colorScheme.error)
                        ) {
                            Text("Delete from Database")
                        }
                    }
                }
            }
        }
    }
}

File: app/ios/image_toolkit/ui/screen/DatabaseScreen.swift
import SwiftUI

struct DatabaseScreen: View {
    @State private var host = "localhost"
    @State private var port = "5432"
    @State private var user = "postgres"
    @State private var password = ""
    @State private var dbName = "image_db"
    
    @State private var imagePath = ""
    @State private var seriesName = ""
    @State private var characters = ""
    @State private var selectedTags: Set<String> = []
    
    @State private var showingAlert = false
    @State private var alertMessage = ""
    
    let allTags = [
        "landscape", "night", "day", "indoor", "outdoor", "solo", "multiple", "fanart",
        "official", "cosplay", "portrait", "full_body", "action", "close_up", "nsfw"
    ]
    
    let seriesOptions = ["Naruto", "Bleach", "One Piece", "Dragon Ball"]
    
    var body: some View {
        ScrollView {
            VStack(alignment: .leading, spacing: 16) {
                // Section 1: Postgres Connection
                SectionCard(title: "PostgreSQL Connection", startOpen: true) {
                    VStack(spacing: 8) {
                        inputField(label: "Host", text: $host)
                        inputField(label: "Port", text: $port, keyboardType: .numberPad)
                        inputField(label: "User", text: $user)
                        
                        VStack(alignment: .leading, spacing: 8) {
                            Text("Password")
                                .font(.caption)
                                .foregroundColor(.gray)
                            SecureField("", text: $password)
                                .textFieldStyle(RoundedBorderTextFieldStyle())
                        }
                        
                        inputField(label: "Database Name", text: $dbName)
                        
                        Button(action: {
                            alertMessage = "Connecting to \(dbName)..."
                            showingAlert = true
                        }) {
                            HStack {
                                Image(systemName: "power")
                                Text("Connect to PostgreSQL")
                            }
                            .frame(maxWidth: .infinity)
                            .padding()
                            .background(AppTheme.primary)
                            .foregroundColor(.white)
                            .cornerRadius(8)
                        }
                    }
                }
                
                Text("Stats: Not connected to database")
                    .frame(maxWidth: .infinity)
                    .padding(16)
                    .background(Color.gray.opacity(0.1))
                    .cornerRadius(8)
                
                // Section 2: Metadata
                SectionCard(title: "Single Image Metadata", startOpen: true) {
                    VStack(spacing: 16) {
                        inputField(label: "Image File Path", text: $imagePath)
                        
                        HStack(spacing: 8) {
                            actionButton(label: "Browse")
                            actionButton(label: "View")
                            actionButton(label: "Load")
                        }
                        
                        ZStack {
                            Rectangle()
                                .fill(AppTheme.secondary.opacity(0.3))
                                .frame(height: 200)
                                .cornerRadius(8)
                            Text("Select an image to edit...")
                                .foregroundColor(AppTheme.onSecondary)
                        }
                        
                        VStack(alignment: .leading, spacing: 8) {
                            Text("Series Name")
                                .font(.caption)
                                .foregroundColor(.gray)
                            
                            Picker("Series Name", selection: $seriesName) {
                                Text("Select Series").tag("")
                                ForEach(seriesOptions, id: \.self) { series in
                                    Text(series).tag(series)
                                }
                            }
                            .pickerStyle(MenuPickerStyle())
                            .frame(maxWidth: .infinity)
                            .padding(8)
                            .background(RoundedRectangle(cornerRadius: 8).stroke(Color.gray.opacity(0.2), lineWidth: 1))
                        }
                        
                        inputField(label: "Characters (comma-separated)", text: $characters)
                        
                        VStack(alignment: .leading, spacing: 8) {
                            Text("Tags")
                                .font(AppTheme.Typography.titleSmall)
                            
                            FlowLayout(items: allTags, spacing: 8) { tag in
                                let isSelected = selectedTags.contains(tag)
                                Button(action: {
                                    if isSelected { selectedTags.remove(tag) } else { selectedTags.insert(tag) }
                                }) {
                                    Text(tag)
                                        .font(.caption)
                                        .padding(.horizontal, 12)
                                        .padding(.vertical, 8)
                                        .background(isSelected ? AppTheme.secondary.opacity(0.2) : Color.gray.opacity(0.1))
                                        .foregroundColor(AppTheme.onSurface)
                                        .cornerRadius(8)
                                        .overlay(
                                            RoundedRectangle(cornerRadius: 8)
                                                .stroke(isSelected ? AppTheme.secondary : Color.clear, lineWidth: 1)
                                        )
                                }
                            }
                        }
                        
                        VStack(spacing: 8) {
                            actionButton(label: "Add/Update Image Path", color: AppTheme.primary, textColor: .white)
                            actionButton(label: "Update Loaded Metadata", color: AppTheme.primary, textColor: .white)
                            actionButton(label: "Delete from Database", color: .red, textColor: .white)
                        }
                    }
                }
            }
            .padding(16)
        }
        .alert("Database Action", isPresented: $showingAlert) {
            Button("OK", role: .cancel) { }
        } message: {
            Text(alertMessage)
        }
    }
    
    @ViewBuilder
    private func inputField(label: String, text: Binding<String>, keyboardType: UIKeyboardType = .default) -> some View {
        VStack(alignment: .leading, spacing: 8) {
            Text(label)
                .font(.caption)
                .foregroundColor(.gray)
            TextField("", text: text)
                .keyboardType(keyboardType)
                .textFieldStyle(RoundedBorderTextFieldStyle())
        }
    }
    
    @ViewBuilder
    private func actionButton(label: String, color: Color = AppTheme.primary.opacity(0.1), textColor: Color = AppTheme.primary) -> some View {
        Button(action: {}) {
            Text(label)
                .frame(maxWidth: .infinity)
                .padding(.vertical, 12)
                .background(color)
                .foregroundColor(textColor)
                .cornerRadius(8)
        }
    }
}